packages/components/src/components/modal/_modal.scss

```diff
@@ -83,31 +84,98 @@
     transition: transform $duration--moderate-02 motion(exit, expressive);
 
     @include carbon--breakpoint(md) {
-      width: 50%;
+      width: 84%;
       max-width: 768px;
       max-height: 90%;
       height: auto;
     }
 
     @include carbon--breakpoint(lg) {
-      max-height: 80%;
+      width: 60%;
+      max-height: 84%;
+    }
+
+    @include carbon--breakpoint(xlg) {
+      width: 48%;
+    }
+  }
+
+  .#{$prefix}--modal-container--xs {
+    @include carbon--breakpoint(md) {
+      width: 48%;
+    }
+
+    @include carbon--breakpoint(lg) {
+      width: 32%;
+      max-height: 48%;
+    }
+
+    @include carbon--breakpoint(xlg) {
+      width: 24%;
+
+      .#{$prefix}--modal-header,
+      .#{$prefix}--modal-content,
+      .#{$prefix}--modal-content__regular-content {
+        padding-right: rem(16px);
+      }
+    }
+  }
+
+  .#{$prefix}--modal-container--sm {
+    @include carbon--breakpoint(md) {
+      width: 60%;
+    }
+
+    @include carbon--breakpoint(lg) {
+      width: 42%;
+      max-height: 72%;
+    }
+
+    @include carbon--breakpoint(xlg) {
+      width: 36%;
+    }
+  }
+
+  .#{$prefix}--modal-container--lg {
+    @include carbon--breakpoint(md) {
+      width: 96%;
+    }
+
+    @include carbon--breakpoint(lg) {
+      width: 84%;
+      max-height: 96%;
+    }
+
+    @include carbon--breakpoint(xlg) {
+      width: 72%;
     }
   }
 
   .#{$prefix}--modal-header,
   .#{$prefix}--modal-content {
-    padding-right: 25%;
     padding-left: 1rem;
   }
 
   .#{$prefix}--modal-header,
-  .#{$prefix}--modal-footer {
-    flex-shrink: 0;
+  .#{$prefix}--modal-content,
+  .#{$prefix}--modal-content__regular-content {
+    padding-right: 20%;
+  }
+
+  .#{$prefix}--modal-content--with-form {
+    padding-right: 1rem;
+
+    @include carbon--breakpoint(xlg) {
+      padding-right: 1rem; // Override for `.#{$prefix}--modal-content`
```

Comment:
should we use a spacing token here instead?

Reply:
Please see: https://github.com/carbon-design-system/carbon/pull/4657/files#r345546565

Reply:
I was referring to the `carbon--spacing` tokens, I think we can use `$spacing-05` right?